scripts/pipeline/orthology_benchmark.py

```diff
@@ -377,6 +379,38 @@ def extract_paralogs(res_dir: str, species_key: str) -> List[Tuple[str, str]]:
     return paralogs
 
 
+def read_in_gtf(species_name: str, gtf_dir: str) -> Union[pandas.DataFrame, type[None]]:
+    """Reads in gene entries from a GTF file for a specified species.
+
+    Args:
+        species_name: Species (genome) name.
+        gtf_dir: Path to the directory with GTF files.
+
+    Returns:
+        A pandas DataFrame with a name of the chromosome or scaffold, start, end and strand for all genes,
+        if the GTF file was found. Otherwise `None`.
+
+    Warns:
+        UserWarning: If the GTF file was not found.
+
+    """
+    gtf_file_pattern = f"{species_name.capitalize()}.*.gtf"
+
+    try:
+        gtf_file = list(glob.glob(os.path.join(gtf_dir, gtf_file_pattern)))[0]
+    except IndexError:
+        warnings.warn(f"GTF file for '{species_name}' not found.")
+        return None
```

Comment:
Is it not dangerous to return a None when there is such error? Why do you not let the exception be raised to the next level?

Reply:
Initially I wasn't returning anything here, just raised a warning. In the subsequent lines, there was an `if` statement to check if there was a `gtf_file`, and if yes, proceeded with parsing and returned a data frame. But `pylint` complained because it wanted a function to always return something, not just when the `if` condition was satisfied. Several rounds of attempts led to this.
 
I will replace that warning with an error. I put a warning to kind of keep it consistent with `get_gtf_file` function where a warning is raised if a GTF file isn't found.